include/sound/sof/topology.h

```diff
@@ -147,6 +148,20 @@ struct sof_ipc_comp_src {
 	uint32_t rate_mask;	/**< SOF_RATE_ supported rates */
 } __packed;
 
+/* generic ASRC component */
```

Comment:
@singalsu this PR looks fine but just wondering why is ASRC not a process type component? That way you wont have to introduce a new IPC struct. And if you do introduce a new IPC struct like here, is it only a minor ABI bump?

Reply:
Yep, I made this RFC exactly for this to get feedback and agree with everyone the approach.

I chose to make it an ASRC instead of process since ASoC already contains the snd_soc_dapm_asrc widget type. To me it makes most sense despite a bit more code than adding new process component.

Reply:
With new kernel and old firmware there would be IPC fail when attempting to load ASRC component. With old kernel and new firmware there would be topology parse fail in kernel. With existing features of SOF no incompatibility. It's not clear to me if this need a major version bump. What do you think?

Reply:
The ASRC component does not exist in old firmware, so what would be the issue with 'new kernel, old firmware'?

Reply:
@plbossart Yep, if we assume used topologies are from the old firmware release then no issues. And with new topology that would use ASRC it would fail with errors reported into kernel log.